src/container/RelayRootContainer.js

```diff
@@ -141,6 +141,28 @@ class RelayRootContainer extends React.Component {
   _runQueries(
     {Component, forceFetch, refetchRoute, route}: RootContainerProps
   ): RootContainerState {
+    var fragmentNames = Component.getFragmentNames();
+    if (__DEV__) {
+      var unmatchedNameSet = {};
+      Object.keys(route.queries).forEach((name) => unmatchedNameSet[name] = 1);
+      fragmentNames.forEach(fragmentName => {
+        // XOR both sets to find unmatched names
+        if (unmatchedNameSet[fragmentName]) {
+          delete unmatchedNameSet[fragmentName];
+        } else {
+          unmatchedNameSet[fragmentName] = 1;
+        }
+      });
+      var unmatchedNames = Object.keys(unmatchedNameSet);
+      invariant(
```

Comment:
Are we dead set on requiring a Relay.Route to include all of a container's fragments, and not just vice versa? It seems a little overly strict to me, and not dealing with that would make the change simpler.